LoopTestingKit/TestingScenarioInstance.swift

```diff
@@ -6,16 +6,58 @@
 //  Copyright © 2019 LoopKit Authors. All rights reserved.
 //
 
+import Foundation
 import LoopKit
 
-public struct TestingScenarioInstance {
+public struct TestingScenarioInstance: RawRepresentable {
```

Comment:
Is the persistence being used in this PR? If not, what is is being added for?

Reply:
Yeah it's used to store the scenario data when the app is quit. When the user re-opens the app it'll restart the scenario. I was also exploring ideas on how to have it continue but didn't come across a good solution.

### `MockCGMDataSource.swift`
<img width="1802" alt="Screenshot 2023-06-27 at 3 42 47 PM" src="https://github.com/tidepool-org/LoopKit/assets/2769158/8f018d15-f5d4-4ac9-b1c7-490e054b3d6c">

Reply:
That example is just persisting samples, right? I'm asking about the full TestingScenarioInstance being made `RawRepresentable`

Reply:
Ahh yes. At one point I persisted the whole instance but did resort to just the glucose entries. I'll update this!

Reply:
Done